src/toil/jobGraph.py

```diff
@@ -148,6 +148,10 @@ def setupJobAfterFailure(self, config, exitReason=None):
         # Set the default memory to be at least as large as the default, in
         # case this was a malloc failure (we do this because of the combined
         # batch system)
+        if exitReason == BatchJobExitReason.MEMLIMIT and config.doubleMem:
+            self._memory = self.memory * 2
+            logger.warning("We have doubled the memory of the failed job %s to %s bytes due to doubleMem flag",
+                           str(self), str(self.memory))
         if self.memory < config.defaultMemory:
```

Comment:
It looks like memory can be both doubled and upped to the default in the same call. That might confuse people; it should do only one or the other.

Reply:
This is true, if the default is greater than double the step's memory, it will be bumped to the default's memory. I actually thought that behavior seemed okay. What will end up happening if the memory is like 500M and the default is 2G and a job fails due to memory:

1a. 500M -> 1G from the memory double
1b. 1G -> 2G due to default
 Job Fails again due to memory:
2a. 2G -> 4G
Job Fails again:
3a. 4G -> 8G
...
I can make these exclusive conditions if you think that would be better.